### PR TITLE
Use Python 3 Pathlib standards, increased error handling

### PR DESCRIPTION
**Pull request by yousefissa**
This PR focuses on enhancing the file operations module by implementing the following changes:

* Replacing the `os` and `os.path` modules with the `pathlib module`, which simplifies file and directory operations while providing better readability.
* Improving exception handling by using specific exceptions (e.g., FileNotFoundError) instead of generic exceptions where appropriate. This change enables more informative error messages to be returned to the user.
* Updating error message formatting by using f-strings, which enhances readability and consistency.

These updates contribute to a more robust and maintainable file operations module, making it easier to understand and modify in the future.

### CODE REVIEW

**Review by nponeccop — CHANGES_REQUESTED** [review]
Rebase against the current master.

File: scripts/constants.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (AndresCdo) [inline scripts/constants.py:1]: Unnecessary file.

File: scripts/execute_code.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
 import docker
-import os
+from docker import errors
```
Comment (AndresCdo) [inline scripts/execute_code.py:2]: Changes not related with PR description.

File: scripts/execute_code.py [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+from pathlib import Path
+from constants import WORKSPACE_DIR
 
 
 def execute_python_file(file):
-    workspace_folder = "auto_gpt_workspace"
+    workspace_folder = WORKSPACE_DIR
```
Comment (AndresCdo) [inline scripts/execute_code.py:8]: Changes not related with PR description.

File: scripts/execute_code.py [hunk 4]
```diff
@@ -38,10 +40,9 @@ def execute_python_file(file):
         logs = container.logs().decode('utf-8')
         container.remove()
 
-        # print(f"Execution complete. Output: {output}")
-        # print(f"Logs: {logs}") 
-
         return logs
 
+    except docker.errors.APIError as e:
```
Comment (AndresCdo) [inline scripts/execute_code.py:45]: Changes not related with PR description.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
An open draft PR from 3 April, oof...

This PR is too far out of date to be fixed up or merged. Since it's also a draft, I don't feel too bad closing it. Still, please feel welcome to join our dev sessions on Discord and to contribute again in the future. :)

**Pull request closed without merging**